src/librustc_mir/transform/elaborate_drops.rs

```diff
@@ -96,15 +96,18 @@ fn find_dead_unwinds<'a, 'tcx>(
                            MaybeInitializedLvals::new(tcx, mir, &env),
                            |bd, p| &bd.move_data().move_paths[p]);
     for (bb, bb_data) in mir.basic_blocks().iter_enumerated() {
-        match bb_data.terminator().kind {
+        let location = match bb_data.terminator().kind {
             TerminatorKind::Drop { ref location, unwind: Some(_), .. } |
-            TerminatorKind::DropAndReplace { ref location, unwind: Some(_), .. } => {
+            TerminatorKind::DropAndReplace { ref location, unwind: Some(_), .. } => location,
+            _ => continue,
+        };
+
                 let mut init_data = InitializationData {
```

Comment:
indentation

Reply:
This is a refactor which is useful for generators with argument.